utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/rules/DualScreenTestRule.kt

```diff
@@ -66,22 +70,31 @@ class DualScreenTestRule : TestRule {
         val dualScreenTestAnnotation = getAnnotation(description, DualScreenTest::class.java)
         val singleScreenTestAnnotation = getAnnotation(description, SingleScreenTest::class.java)
         val deviceOrientationAnnotation = getAnnotation(description, DeviceOrientation::class.java)
+        val mockFoldingFeatureAnnotation = getAnnotation(description, MockFoldingFeature::class.java)
         val requestedDeviceOrientation = requestedDeviceOrientation(
             dualScreenTestAnnotation,
             singleScreenTestAnnotation,
             deviceOrientationAnnotation
         )
 
         val runDualScreenTest = dualScreenTestAnnotation != null
-        if (uiDevice.isSurfaceDuo()) {
-            if (runDualScreenTest) {
-                uiDevice.spanFromStart()
+        when {
+            mockFoldingFeatureAnnotation != null -> {
+                WindowInfoTracker.overrideDecorator(overrideServices)
+                mockFoldingFeature(mockFoldingFeatureAnnotation)
+            }
+
+            uiDevice.isSurfaceDuo() -> {
+                if (runDualScreenTest) {
+                    uiDevice.spanFromStart()
+                }
             }
-        } else {
-            WindowInfoTracker.overrideDecorator(overrideServices)
-            mockFoldingFeature(requestedDeviceOrientation, runDualScreenTest)
-        }
 
+            else -> {
+                WindowInfoTracker.overrideDecorator(overrideServices)
+                mockFoldingFeature(requestedDeviceOrientation, runDualScreenTest)
+            }
+        }
```

Comment:
I want to keep that @SingleScreenTest/@DualScreenTest only for this kind of test, when the app is in single mode and dual screen. If I'll do this: @DualScreenTest(mockFoldingFeature = MockFoldingFeature(windowBounds = ..., center = ..., size = ..., state = ..., orientation = ...),then I can add there whatever I want and may destroy some tests, this was the reason that I added a separate annotation. @DualScreenTest will mock the folding feature only for other foldable devices, and will mock it with something simillar with the real device.

Reply:
I see, that makes sense! If I understand correctly, that means that developers should use either @DualScreenTest or @MockFoldingFeature, not both at once right?

In either case, it may be good to just add some checks to make sure developers aren't accidentally calling multiple conflicting annotations at the same time - I'm not sure if dual/single screen would take priority over the mock folding feature, but whatever you choose should be fine as long as it's well documented 😄

Reply:
Yeah, not both of them on the same time. I'll add this check on the runner class and also I'll write this on the documentation. Thanks

Reply:
Done